src/app/store/machine/utils/hooks.ts

```diff
@@ -181,6 +182,7 @@ export const useFetchMachine = (
   const machine = useSelector((state: RootState) =>
     machineSelectors.getById(state, id)
   );
+  const previousMachine = usePreviousPersistent(machine);
```

Comment:
I think the machine select is the only case we have right now that needs the previous data. I wonder if we should move this to `MachineSelect` until we have more than one use case? Not really sure, it's probably not that costly to compute this, it just feels like it fits this use case, but might not fit all cases (it's not really keeping the previous data, it's storing a local copy, but the data will be removed from the store so can't be selected).

Reply:
I am fairly sure it will be useful for other use cases in the future, but went with your suggestion and moved this to `MachineSelect` until that actually happens.